src/frontend/komponenter/ManuellJournalføring/ManuellJournalføring.tsx

```diff
@@ -0,0 +1,232 @@
+            });
+        }
+
+        if (dokumentType === '') {
+            accFeilmeldinger.push({
+                feilmelding: 'Du må sette dokumenttype for dokumentet',
+                skjemaelementId: 'manuell-journalføring-dokumenttype',
+            });
+        }
+
+        if (annetInnhold === '') {
+            accFeilmeldinger.push({
+                feilmelding: 'Du må sette annet innhold for dokumentet',
+                skjemaelementId: 'manuell-journalføring-annet-innhold',
+            });
+        }
+
+        settFeilmeldinger(accFeilmeldinger);
+        return accFeilmeldinger;
+    };
+
+    switch (dataForManuellJournalføring.status) {
+        case RessursStatus.IKKE_HENTET:
+        case RessursStatus.HENTER:
+            return <SystemetLaster />;
+        case RessursStatus.SUKSESS:
+            return dataForManuellJournalføring.data.journalpost.journalstatus ===
+                Journalstatus.MOTTATT ? (
+                <Skjemasteg
+                    className={'journalføring'}
+                    tittel={'Registrere journalpost: Barnetrygd'}
+                    forrigeKnappTittel={'Avbryt'}
+                    forrigeOnClick={() => {
+                        history.push(`/oppgaver`);
+                    }}
+                    nesteKnappTittel={'Journalfør'}
+                    nesteOnClick={() => {
+                        const accFeilmeldinger = validerSkjema();
+
+                        if (
+                            accFeilmeldinger.length === 0 &&
+                            person.status === RessursStatus.SUKSESS
+                        ) {
+                            settSenderInn(true);
+                            axiosRequest<string, IRestOppdaterJournalpost>({
+                                method: 'PUT',
+                                url: `/familie-ba-sak/api/journalpost/${
+                                    dataForManuellJournalføring.data.journalpost.journalpostId
+                                }/ferdigstill/${oppgaveId}?journalfoerendeEnhet=${innloggetSaksbehandler?.enhet ??
+                                    '9999'}`,
+                                data: {
+                                    bruker: {
+                                        navn: person.data.navn,
+                                        id: person.data.personIdent,
+                                    },
+                                    avsender: {
+                                        navn: person.data.navn,
+                                        id: person.data.personIdent,
+                                    },
+                                    mottattDato,
+                                    dokumentType,
+                                    annetInnhold,
+                                    knyttTilFagsak,
+                                },
+                            })
+                                .then((fagsakId: Ressurs<string>) => {
```

Comment:
Pull out this async function to make the jsx more sensible to read.

Reply:
This is actually "my code" and has been reviewed :) This component will change in short time with better css, validation and error messages, so no need to change it :) 

It is functional, but kind of a MVP for manuelle journalføring.